gomock_generator/gomockgenerator/generate.go

```diff
@@ -67,7 +69,12 @@ func GenerateMocks(ctx context.Context, gcfg GenerationConfiguration, mocksCfg M
 	if mocksCfg.BaseDirectory == "" {
 		mocksCfg.BaseDirectory = mocksCfg.BasePackage
 	}
-	err = os.Chdir(path.Join(os.Getenv("GOPATH"), "src", mocksCfg.BaseDirectory))
+
+	mocksPath := path.Join(path.Join(os.Getenv("GOPATH"), "src"), mocksCfg.BaseDirectory)
+	if gcfg.SourcePath != "" {
+		mocksPath = gcfg.SourcePath
+	}
```

Comment:
issue: IMO the check with the argument `SourcePath` should only be done once. The fact that you need to do the check over and over again is a sign that there is a lack of optimization somewhere. Doing this check here, at the beginning of the mocks generation is ok. Doing it multiple times makes me feel like something is wrong. Can you have a look to optimize this?